tests/utils.py

```diff
@@ -248,12 +305,12 @@
 
 def get_keys(query: Union[Query, Post]) -> Dict[str, Any]:
     """Converts dataclass object to dict and returns dict without empty values"""
-    return {key: value for key, value in query._asdict().items() if bool(value)}
+    return {key: value for key, value in asdict(query).items() if bool(value)}
 
 
 def set_values(model, values):
-    """Sets NamedTuple instance value and returns new NamedTuple"""
-    return model.__class__(**{**model._asdict(), **values})
+    """Return a new dataclass object replacing specified fields with new values"""
+    return replace(model, **values)
```

Comment:
I would directly use replace instead of set_values and remove set_values function

Reply:
`set_values` was removed